From d2e07566afd0b6f88f0bac426946b0a5fbbb7ec7 Mon Sep 17 00:00:00 2001
From: Scott Pierce <ddrscott@gmail.com>
Date: Wed, 6 Feb 2013 14:19:52 -0600
Subject: [PATCH] use Apple's flag instead of reinventing the wheel

solution for issue #28
---
 Reachability.h |   22 ++++------------------
 1 file changed, 4 insertions(+), 18 deletions(-)

diff --git a/Reachability.h b/Reachability.h
index 6cce494..9febed8 100644
--- a/Reachability.h
+++ b/Reachability.h
@@ -38,27 +38,13 @@
 /**
  * Does ARC support support GCD objects?
  * It does if the minimum deployment target is iOS 6+ or Mac OS X 8+
+ * 
+ * @see http://opensource.apple.com/source/libdispatch/libdispatch-228.18/os/object.h
  **/
-#if TARGET_OS_IPHONE
-
-// Compiling for iOS
-
-#if __IPHONE_OS_VERSION_MIN_REQUIRED >= 60000 // iOS 6.0 or later
-#define NEEDS_DISPATCH_RETAIN_RELEASE 0
-#else                                         // iOS 5.X or earlier
-#define NEEDS_DISPATCH_RETAIN_RELEASE 1
-#endif
-
-#else
-
-// Compiling for Mac OS X
-
-#if MAC_OS_X_VERSION_MIN_REQUIRED >= 1080     // Mac OS X 10.8 or later
+#if OS_OBJECT_USE_OBJC
 #define NEEDS_DISPATCH_RETAIN_RELEASE 0
 #else
-#define NEEDS_DISPATCH_RETAIN_RELEASE 1     // Mac OS X 10.7 or earlier
-#endif
-
+#define NEEDS_DISPATCH_RETAIN_RELEASE 1
 #endif
 
 
-- 
1.7.10

